src/pin.rs

```diff
@@ -0,0 +1,203 @@
+//! Mock digital InputPin and OutputPin implementations
+//! 
+//! ```
+//! use std::io::ErrorKind;
+//! 
+//! use embedded_hal_mock::MockError;
+//! use embedded_hal_mock::pin::{Transaction as PinTransaction, Mock as MockPin};
+//! use embedded_hal::digital::v2::{InputPin, OutputPin};
+//! 
+//! // Configure expectations
+//! let expectations = [
+//!     PinTransaction::get(true),
+//!     PinTransaction::get(true),
+//!     PinTransaction::set(false),
+//!     PinTransaction::set(true).with_error(MockError::Io(ErrorKind::NotConnected)),
```

Comment:
Do you think it would be possible to make this line somehow shorter?

Reply:
We could impl `From<io::ErrorKind> for MockError` and use `into()`, but I think I'd actually prefer the explicit version.

Reply:
Hmm, I just need an `Error` instance, it doesn't have to me a `MockError` it was just conveniently there.

I could swap to `io::Error::new(ErrorKind::Other, "whoops")` or something, but it's not measurably shorter.

Reply:
Oh wait yes I do. Could make a `MockError::not_connected()` helper or something?

Reply:
But then we'd have to add a helper for every possible `ErrorKind`, right? I doubt that this is a good general solution.

In real code, if you re-use the same error a lot, you can always create a `MockError` instance and simply clone it every time you need it:

```rust
let not_connected = MockError::Io(ErrorKind::NotConnected);
...
PinTransaction::set(true).with_error(not_connected.clone())
```

Reply:
@dbrgn that solution is probably enough. I also added a suggestion of making the error types copyable (see below) which would further simplify this.